### PR TITLE
Fix compilation error on new OpenBMC codebase

### PR DESCRIPTION
**Pull request by hieuhuynh-ampere**
Fix compilation error on new OpenBMC codebase by correct tab on somes
code to fix "this ‘if’ clause does not guard... [-Wmisleading-indentation]"
warning.
This commit also correct the size of variables to fix
"[-Werror=stringop-overflow=]" error.

Signed-off-by: Hieu Huynh <hieuh@os.amperecomputing.com>

### CODE REVIEW

**Review by thangqn-ampere — APPROVED** [review]

**Review by dgoehring-ampere — CHANGES_REQUESTED** [review]
Some issues need to be addressed...
1. The openocd.bb file that OpenBMC uses is located at:
openbmc/meta-ampere/meta-jade/recipes-ampere/openocd/openocd.bb

The file references the "ampere_openocd_v1.2.0" branch, not the AmpereComputing:master branch. To address this fix, a new branch needs to be created.

From openocd.bb...
#Remote Git Repository
SRC_URI = "git://github.com/AmpereComputing/ampere-openocd.git;protocol=https;branch=ampere_openocd_v1.2.0"

The openocd.bb file also hardcodes the SRCREV, therefore updating ampere_openocd_v1.2.0 won't resolve the issue. A new branch needs to be created along with update SRCREV.
SRCREV = "32a2a48bbf323977bc09f720476fcfbad0ee46be"

2. I think the best course of action is to pick up the latest OpenOCD files which include a gcc11 fix.
https://sourceforge.net/p/openocd/code/ci/6db49eb8858023654084088c73d968e190aec491/

In summary, while this patch addresses the gcc11 build issue, it can't be applied to this branch and the openocd.bb file needs to be updated to reflect the new branch and SRCREV. Best option going forward is for me to update OpenOCD with the latest source which includes the gcc11 fix.

What is the timeframe that a new OpenOCD drop needs to be available?
